### ISSUE TITLE
Option to make Scan working with bosses

### ISSUE THREAD

**Issue by Gianluca90** [Albeoris/Memoria#981]
Hello,
this is not an issue, but rather a suggestion. 
Is it possibile to put an option in the launcher which make Scan always work even with bosses? A tick-box similar to the "100% steal rate" would be enough.
Or if that's not in your plans, what i need to change in the Scan script to make it always work, even with bosses?
Thank you.

**Comment by Tirlititi** [comment]
Hello.
The problem is that many enemies and bosses that are immune to Scan have a weird HP counter. Many of them have 10,000 HP more than what needs to be dealt to end the fight, and some of them have yet another system.

You can use a custom C# script for Scan to remove the immunity, but then you'll see strange HP values sometimes:
```csharp
using System;

namespace Memoria.Scripts.Battle
{
    /// <summary>
    /// Scan
    /// </summary>
    [BattleScript(Id)]
    public sealed class ScanScript : IBattleScript
    {
        public const Int32 Id = 0059;

        private readonly BattleCalculator _v;

        public ScanScript(BattleCalculator v)
        {
            _v = v;
        }

        public void Perform()
        {
            // The condition "_v.Target.CheckUnsafetyOrMiss()" is removed
            _v.Target.Libra();
        }
    }
}
```

**Comment by Gianluca90** [comment]
Thanks @Tirlititi for the answer. 
So basically i have to just remove that line; if i recall correctly, to edit a script i have to:

- open it in StreamingAssets\Scripts\Sources\Battle
- edit it
- run the compiler in StreamingAssets\Scripts\Compiler\Memoria.Compiler.exe 
- profit (???)

Is it right? It's been a while since last time.
Also, is there a way ti see enemies HP without those 10k more hp? Or is it hardcoded in the game?
Thanks again

**Comment by Tirlititi** [comment]
That a way to do it, yes. When you edit C# scripts like that, it's harder to go back and undo the change. Since there's only 1 line that changes there, it's not really a problem. Otherwise, you'd do it like [explained here](https://github.com/Albeoris/Memoria/wiki/External-C%23-scripts).

The main problems with the 10k more HP are that it's complicated to identify which enemies indeed have these 10k extra HP (sometimes, it changes mid-battle, like with Plant Brain) and that it's not always 10k, but sometimes a variable amount of extra HP.
It's not really hard-coded, it's just their real HP.

Maybe at some point it will be easier to deal with that system, but it's not the case for now.

**Comment by Gianluca90** [comment]
@Tirlititi Got it, thanks. I gave a quick look to the link you posted, but i didn't understand much :D i think i'll just remove the line from the Scan script and call it a day.
Just for curiosity, why you say that's harder to undo the changes? Isn't just a matter of making a copy of the original script, and compile it back in case something goes wrong? Or am i missing something?
Thanks

**Comment by Tirlititi** [comment]
Yes it's that. When it concerns plenty of files that you didn't keep track of, it can be tedious (while using a mod folder is just a matter of enabling / disabling the mod on the launcher).

**Comment by Gianluca90** [comment]
> Yes it's that. When it concerns plenty of files that you didn't keep track of, it can be tedious (while using a mod folder is just a matter of enabling / disabling the mod on the launcher).

Cool, thanks again.
Last question: if i have questions about scripts editing or stuff like that, what can i do? I don't want to post those things here in the issues and flood the sections. Is there a discord or something?
Thanks

**Comment by Tirlititi** [comment]
There's the wiki: https://github.com/Albeoris/Memoria/wiki/External-Battle-scripts
And there's indeed a discord many modders are in, although it's not the official discord or whatever: https://discord.gg/b4D9SRG9

**Comment by Gianluca90** [comment]
> There's the wiki: https://github.com/Albeoris/Memoria/wiki/External-Battle-scripts And there's indeed a discord many modders are in, although it's not the official discord or whatever: https://discord.gg/b4D9SRG9

Thank you very much for all the information, you've been very helpful.
I'll mark the issue as solved.
